packages/search-metadata-previews/src/snippet-preview/SnippetPreview.js

```diff
@@ -732,20 +732,25 @@ export default class SnippetPreview extends PureComponent {
 	renderProductData() {
 		const {	mode, shoppingData } = this.props;
 
-		if ( mode === MODE_DESKTOP && shoppingData !== null ) {
-			return (
-				<ProductDataDesktop
-					shoppingData={ shoppingData }
-				/>
-			);
-		} else if ( mode === MODE_MOBILE && shoppingData !== null ) {
-			return (
-				<ProductDataMobile
-					shoppingData={ shoppingData }
-				/>
-			);
-		}
-		return null;
+		if ( shoppingData !== {} ) {
+
+			if ( mode === MODE_DESKTOP ) {
+				return (
+					<ProductDataDesktop
+						shoppingData={ shoppingData }
+					/>
+				);
+			}
+			
+			if ( mode === MODE_MOBILE ) {
+				return (
+					<ProductDataMobile
+						shoppingData={ shoppingData }
+					/>
+				);
+			}
+
+			return null;	
```

Comment:
A couple of things:
- You can not compare an object to an empty different object. It compares the memory address and thus will never be the same. You can work around this by checking if there are any values for example.
- You can early return on the shoppingData check so you don't have to indent for all the rest. This improves readability of your code.
- You are missing a return for when the shoppingData is empty, that will be `undefined` instead of `null` in your current code. That is not a problem in behavior, but it is not correct.

```suggestion
		if ( Object.values( shoppingData ).length === 0 ) {
			return null;
		}
	
		if ( mode === MODE_DESKTOP ) {
			return (
				<ProductDataDesktop
					shoppingData={ shoppingData }
				/>
			);
		}
	
		if ( mode === MODE_MOBILE ) {
			return (
				<ProductDataMobile
					shoppingData={ shoppingData }
				/>
			);
		}
	
		return null;
```